pallets/subtensor/src/staking.rs

```diff
@@ -373,4 +374,23 @@ impl<T: Config> Pallet<T> {
         };
     }
 
+    pub fn unstake_all_coldkeys_from_hotkey_account( hotkey: &T::AccountId ) {
+        // Iterate through all coldkeys that have a stake on this hotkey account.
+        for ( delegate_coldkey_i, stake_i ) in < Stake<T> as IterableStorageDoubleMap<T::AccountId, T::AccountId, u64 >>::iter_prefix( hotkey ) {
+            
+            // Convert to balance and add to the coldkey account.
+            let stake_i_as_balance = Self::u64_to_balance( stake_i );
+            if stake_i_as_balance.is_none() {
+                continue; // Don't unstake if we can't convert to balance.
+            } else {
+                // Stake is successfully converted to balance.
+                // Add the balance to the coldkey account.
+                Self::add_balance_to_coldkey_account( &delegate_coldkey_i, stake_i_as_balance.unwrap() );
```

Comment:
The use of `stake_i_as_balance` here vs `stake_i` further on (at line `391` has be a bit concerned.

> Also the order of operations may be better if reversed, though this could be my past JavaScript trauma talking here

Reply:
> Swapped the statements

Thank you!

> I'm not sure there's a better way to do that. afaik this is how we do it elsewhere, because the stake map is a `u64` and the balances are a `Balance`

:thinking: oof, that is a bit troubling from my perspective...  Initially I thought, "why not cast `Balance` to `u64`", but diving into the underlying source for the `Balance` shows we may not be guaranteed loss-less conversion;

- `<substrate-cache>/primitives/arithmetic/src/traits.rs:151`

```rust
/// A meta trait for arithmetic.
///
/// Arithmetic types do all the usual stuff you'd expect numbers to do. They are guaranteed to
/// be able to represent at least `u32` values without loss, hence the trait implies `From<u32>`
/// and smaller integers. All other conversions are fallible.
pub trait AtLeast32Bit: BaseArithmetic + From<u16> + From<u32> {}
```

- `<substrate-cache>/primitives/arithmetic/src/traits.rs:161`

```rust
/// A meta trait for arithmetic.  Same as [`AtLeast32Bit `], but also bounded to be unsigned.
pub trait AtLeast32BitUnsigned: AtLeast32Bit + Unsigned {}
```

... Regardless this is smelling like a _yak shaving_ sort of thing, so I'll totally trust your implementation is solid.